### PR TITLE
Added adapter for nanodet models

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py [hunk 1]
```diff
@@ -933,3 +895,101 @@ def process(self, raw, identifiers, frame_meta):
...
+        return parameters
+
+    def configure(self):
+        self.num_classes = self.get_value_from_config('num_classes')
+        self.confidence_threshold = self.get_value_from_config('confidence_threshold')
+        self.nms_threshold = self.get_value_from_config('nms_threshold')
+        self.max_detections = self.get_value_from_config('max_detections')
+        # Set default values
+        self.strides = [8, 16, 32]
+        self.reg_max = 7
```
Comment (eaidova) [inline tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py:929]: are there situation when this value can be changed? Possibly we should make them also configurable, but with default parameters

File: tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py [hunk 2]
```diff
@@ -933,3 +895,101 @@ def process(self, raw, identifiers, frame_meta):
...
+        list_center_priors = []
+        for i, stride in enumerate(self.strides):
+            y, x = self.get_single_level_center_point(featmap_sizes[i], stride)
+            strides = np.full_like(x, stride)
+            list_center_priors.append(np.stack([x, y, strides, strides], axis=-1))
+        center_priors = np.concatenate(list_center_priors, axis=0)
+        dist_project = np.linspace(0, self.reg_max, self.reg_max + 1)
+        x = np.dot(softmax(np.reshape(reg_preds, (*reg_preds.shape[:-1], 4, self.reg_max + 1)), -1), dist_project)
+        dis_preds = x * np.expand_dims(center_priors[:, 2], -1)
+        return self.distance2bbox(center_priors[:, :2], dis_preds, (input_height, input_width))
```
Comment (eaidova) [inline tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py:957]:
```suggestion
        for stride, featmap_size in zip(self.strides, featmap_sizes):
            y, x = self.get_single_level_center_point(featmap_sizes, stride)
            strides = np.full_like(x, stride)
            list_center_priors.append(np.stack([x, y, strides, strides], axis=-1))
        center_priors = np.concatenate(list_center_priors, axis=0)
        dist_project = np.linspace(0, self.reg_max, self.reg_max + 1)
        x = np.dot(softmax(np.reshape(reg_preds, (*reg_preds.shape[:-1], 4, self.reg_max + 1)), -1), dist_project)
        dis_preds = x * np.expand_dims(center_priors[:, 2], -1)
        return self.distance2bbox(center_priors[:, :2], dis_preds, (input_height, input_width))
```

File: tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py]: Done

File: tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py (hunk 2 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/detection.py]: Done

**Review by eaidova — APPROVED** [review]

**Pull request merged**